scripts/bumpfee.py

```diff
@@ -0,0 +1,254 @@
+    tx_fee = btc.amount_to_sat(abs(tx_info['fee']))
+    proposed_fee_rate = fee_rate / Decimal(1000.0)
+    proposed_fee = int(tx_size_n * proposed_fee_rate)
+    min_proposed_fee = tx_fee + tx_size_n
+    min_proposed_fee_rate = min_proposed_fee / Decimal(tx_size_n)
+
+    if proposed_fee < (tx_fee + tx_size_n):
+        raise ValueError('Proposed fee for transaction replacement: '
+                         '%d sats (%.1f sat/vB) is below minimum required '
+                         'for relay: %d sats (%.1f sat/vB). '
+                         'Try using a higher fee setting.'
+                         % (proposed_fee, proposed_fee_rate, 
+                            min_proposed_fee, min_proposed_fee_rate))
+
+    return (proposed_fee - tx_fee)
+
+def prepare_transaction(new_tx, old_tx, wallet):
+    input_scripts = {}
+    spent_outs = []
+    for ix, vin in enumerate(new_tx.vin):
+        script = wallet.pubkey_to_script(
+            btc.extract_pubkey_from_witness(old_tx, ix)[0])
+        tx_info = jm_single().bc_interface.get_transaction(
+            hextobin(btc.b2lx(new_tx.vin[ix].prevout.hash)))
+        prev_tx = btc.CTransaction.deserialize(hextobin(tx_info['hex']))
+        amount = prev_tx.vout[new_tx.vin[ix].prevout.n].nValue
+
+        input_scripts[ix] = (script, amount)
+        spent_outs.append(btc.CMutableTxOut(amount, script))
+
+    return (input_scripts, spent_outs)
+
+def sign_transaction(new_tx, old_tx, wallet_service):
+    input_scripts, _ = prepare_transaction(new_tx, old_tx, wallet_service.wallet)
+    success, msg = wallet_service.sign_tx(new_tx, input_scripts)
+    if not success:
+        raise RuntimeError("Failed to sign transaction, quitting. Error msg: " + msg)
+
+def sign_psbt(new_tx, old_tx, wallet_service):
+    _, spent_outs = prepare_transaction(new_tx, old_tx, wallet_service.wallet)
+    unsigned_psbt = wallet_service.create_psbt_from_tx(
+        new_tx, spent_outs=spent_outs)
+    print(unsigned_psbt.to_base64())
+    signed_psbt, err = wallet_service.sign_psbt(unsigned_psbt.serialize())
+
+    if err:
+        raise RuntimeError("Failed to sign PSBT, quitting. Error message: " + err)
+
+    return btc.PartiallySignedTransaction.deserialize(signed_psbt)
+
+def create_bumped_tx(tx, wallet, fee):
+    for ix, vout in enumerate(tx.vout):
+        if wallet.is_known_script(vout.scriptPubKey):
+            # check if the output is a change address
+            if wallet.script_to_path(vout.scriptPubKey)[-2] != 1:
+                continue
+
+            # deduct fee from the change
+            tx.vout[ix].nValue -= fee
+
+            # if the output value is below the dust threshold, remove it
+            if tx.vout[ix].nValue <= jm_single().BITCOIN_DUST_THRESHOLD:
+                jlog.info("Dynamically calculated change lower than dust: " +
+                    btc.amount_to_str(tx.vout[ix].nValue) + "; dropping.")
+                tx.vout.remove(vout)
+            break
+
```

Comment:
I think you need to check if any change output was found; if not we can't bump, right? Currently I think it'll just return the same tx which is not what we want?

Reply:
The check is already performed by `check_valid_candidate`.